hack/test/e2e.sh

```diff
@@ -102,12 +102,17 @@ function run_talos_integration_test_docker {
 }
 
 function run_kubernetes_integration_test {
-  ${SONOBUOY} run \
+  timeout=$(($(date +%s) + ${TIMEOUT}))
+  until ${SONOBUOY} run \
     --kubeconfig ${KUBECONFIG} \
     --wait \
     --skip-preflight \
     --plugin e2e \
-    --mode ${SONOBUOY_MODE}
+    --mode ${SONOBUOY_MODE}; do
+    [[ $(date +%s) -gt $timeout ]] && exit 1
+    echo "attempting to run sonobuoy"
+    sleep 10
+  done  
```

Comment:
In the case of running in the certification mode, will this run through the whole thing again if it fails? It looks it actually covers the case when running the sonobuoy command fails initially at all for some reason. Correct me if I am wrong.

Reply:
Yeah, it'll re-kick if sonobuoy fails at all initially. But the way I'm thinking it will work is, since sonobuoy takes forever to run, if it fails deep in the run for some reason, it'd be past the timeout and would exit w/ error code 1.